libs/wingsdk/src/std/resource.ts

```diff
@@ -506,6 +518,16 @@ export class Display {
    */
   public description?: string;
 
+  /**
+   * Whether the resource is user-named or compiler-named.
+   */
+  public type?: "user-named" | "compiler-named";
```

Comment:
I feel like the generalized question of what we're trying get at is, who authored the class? Was it the user creating an app? Was it their teammate? Was it the author of the library? Was it us (Wing Cloud Library maintainers)?

The most robust way to get this answer is based on where the class was defined. Right now Wing SDK is the only library in Wing's ecosystem, but in the future there will be others. I'm thinking perhaps we could simply store the name of the library / file where the class was defined, like so:

```suggestion
  /**
   * The source file or library where the resource was defined.
   */
  public sourceModule?: string;
```

This could take on the name of a library, or the name of a source file, like "@winglang/sdk", "wing-redis", "main.w", etc.

Reply:
I replaced the `type` attribute for `sourceModule` and set it as `@winglang/sdk` for the Functions generated by the SDK. 

Let me know if you have a better idea!